src/cim/components/featureSupportLevels/ACMFeatureSupportLevelProvider.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { PropsWithChildren } from 'react';
+
+import { getOCPVersions } from '../helpers';
+import {
+  FeatureSupportLevelsMap,
+  FeatureIdToSupportLevel,
+  FeatureId,
+  SupportLevel,
+  FeatureSupportLevelContextProvider,
+  FeatureSupportLevelData,
+} from '../../../common';
+import { ClusterImageSetK8sResource } from '../../types';
+
+export type ACMFeatureSupportLevelProvider = PropsWithChildren<{
+  clusterImages: ClusterImageSetK8sResource[];
+}>;
+
+export const ACMFeatureSupportLevelProvider: React.FC<ACMFeatureSupportLevelProvider> = ({
+  children,
+  clusterImages,
+}) => {
+  const ocpVersions = React.useMemo(() => getOCPVersions(clusterImages), [clusterImages]);
+
+  const supportLevelData: FeatureSupportLevelsMap = React.useMemo<FeatureSupportLevelsMap>(() => {
+    const featureSupportLevelsMap: FeatureSupportLevelsMap = {};
+
+    ocpVersions.forEach((version) => {
+      // TODO(mlibra): get this dynamically
+      const featureIdToSupportLevel: FeatureIdToSupportLevel = {
+        REQUESTED_HOSTNAME: 'supported',
+        SNO: 'supported',
+        CNV: 'supported',
+        NETWORK_TYPE_SELECTION: 'supported',
+      };
+      featureSupportLevelsMap[version.version] = featureIdToSupportLevel;
+    });
+    return featureSupportLevelsMap;
+  }, [ocpVersions]);
+
+  const getVersionSupportLevelsMap: FeatureSupportLevelData['getVersionSupportLevelsMap'] = React.useCallback(
+    (versionName: string): FeatureIdToSupportLevel | undefined => {
+      const version = ocpVersions.find((v) => v.value == versionName);
+      const normalized = version?.version;
+      return normalized
+        ? supportLevelData[normalized]
+        : {
+            /* empty FeatureIdToSupportLevel */
+          };
+    },
+    [supportLevelData, ocpVersions],
+  );
+
+  // TODO(mlibra): Following callbacks can be reused with the OCM flow, just based on providing an application-specific map
+  const getFeatureSupportLevel: FeatureSupportLevelData['getFeatureSupportLevel'] = React.useCallback(
+    (versionName: string, featureId: FeatureId): SupportLevel | undefined => {
+      const versionSupportLevelData = getVersionSupportLevelsMap(versionName);
+      return versionSupportLevelData ? versionSupportLevelData[featureId] : undefined;
+    },
+    [getVersionSupportLevelsMap],
+  );
+
+  const isFeatureSupportedCallback: FeatureSupportLevelData['isFeatureSupported'] = React.useCallback(
+    () => true,
+    [],
+  );
+
+  const getDisabledReasonCallback: FeatureSupportLevelData['getFeatureDisabledReason'] = React.useCallback(() => {
+    return undefined; // we have recently everything enabled in the ACM
```

Comment:
SNO is always enabled in ACM? What about with OCP 4.8?

Reply:
Right. It's weird since it is a TODO. So far there is everything on the "production" support level.
@jelkosz Do we have an ACM 2.5 task for that? Or plan to change it?

Reply:
it should not be the case - Im sure I have seen the banner for sno for older version in acm. But checking now it does not work.
@jtomasek any idea what happened to it?

Reply:
It's since I removed the hard coded behavior of this banner, it needs to be implemented in the featuresupportlevel provider

Reply:
You might need to add to this the logic of SNO being disabled in version < 4.9, but as you said it can be done in a separate task